[Bug]: Storybook 7 not working with pnpm + Next.js 13 + TailwindCSS

### Describe the bug

Adding Storybook 7 to a new app with pnpm + Next.js 13 + TailwindCSS results in this error:

```
ModuleNotFoundError: Module not found: Error: Can't resolve 'style-loader' in '/private/tmp/my-app'
```

### To Reproduce

```sh
pnpm --version
# 7.29.1
pnpm create next-app my-app --ts --eslint --src-dir --no-experimental-app --import-alias '@/*'
cd my-app
pnpm dev # just to check that everything works
pnpm install -D tailwindcss postcss autoprefixer
pnpx tailwindcss init -p
# add "./src/**/*.{js,ts,jsx,tsx}" to `tailwind.config.js`
# add `@tailwind` directives to globals.css
# add `text-5xl` to some tag on the page
pnpm dev # just to check that everything works
pnpx storybook@next init
pnpm storybook # boom, you get the error
```

<img width="1072" alt="Screenshot 2023-03-14 at 9 55 17 AM" src="https://user-images.githubusercontent.com/123422405/225080040-753659a2-477b-49d6-8c94-aa448141d6a5.png">

```
> pnpm storybook

> my-app@0.1.0 storybook /private/tmp/my-app
> storybook dev -p 6006

@storybook/cli v7.0.0-rc.3

info => Starting manager..
WARN You (or an addon) are using the 'config' preset field. This has been replaced by 'previewAnnotations' and will be removed in 8.0
info Addon-docs: using MDX2
info => Using implicit CSS loaders
info => Using default Webpack5 setup
<i> [webpack-dev-middleware] wait until bundle finished
WARN Force closed manager build
99% end closing watch compilationWARN Force closed preview build
ModuleNotFoundError: Module not found: Error: Can't resolve 'css-loader' in '/private/tmp/my-app'
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/Compilation.js:1506:33
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:644:18
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:10:1)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:197:20
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:9:1)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:404:22
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:60:14
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:476:15
    at /private/tmp/my-app/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:694:7
    at done (/private/tmp/my-app/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:771:13)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:831:22
    at finishWithoutResolve (/private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:193:14)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:258:22
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:27:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:72:26
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:15:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:42:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/ConditionalPlugin.js:40:20
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/forEachBail.js:11:16
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/ModulesInHierarchicalDirectoriesPlugin.js:59:22
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
    at runNextTicks (node:internal/process/task_queues:65:3)
    at processImmediate (node:internal/timers:437:9)
resolve 'css-loader' in '/private/tmp/my-app'
  Parsed request is a module
  using description file: /private/tmp/my-app/package.json (relative path: .)
    resolve as module
      looking for modules in /private/tmp/my-app/node_modules
        single file module
          using description file: /private/tmp/my-app/package.json (relative path: ./node_modules/css-loader)
            no extension
              /private/tmp/my-app/node_modules/css-loader doesn't exist
            .js
              /private/tmp/my-app/node_modules/css-loader.js doesn't exist
        /private/tmp/my-app/node_modules/css-loader doesn't exist
      /private/tmp/node_modules doesn't exist or is not a directory
      /private/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
ModuleNotFoundError: Module not found: Error: Can't resolve 'css-loader' in '/private/tmp/my-app'
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/Compilation.js:1506:33
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:644:18
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:10:1)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:197:20
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:9:1)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:404:22
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:60:14
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:476:15
    at /private/tmp/my-app/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:694:7
    at done (/private/tmp/my-app/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:771:13)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:831:22
    at finishWithoutResolve (/private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:193:14)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:258:22
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:27:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:72:26
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:15:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:42:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/ConditionalPlugin.js:40:20
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/forEachBail.js:11:16
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/ModulesInHierarchicalDirectoriesPlugin.js:59:22
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
    at runNextTicks (node:internal/process/task_queues:65:3)
    at processImmediate (node:internal/timers:437:9)
resolve 'css-loader' in '/private/tmp/my-app'
  Parsed request is a module
  using description file: /private/tmp/my-app/package.json (relative path: .)
    resolve as module
      looking for modules in /private/tmp/my-app/node_modules
        single file module
          using description file: /private/tmp/my-app/package.json (relative path: ./node_modules/css-loader)
            no extension
              /private/tmp/my-app/node_modules/css-loader doesn't exist
            .js
              /private/tmp/my-app/node_modules/css-loader.js doesn't exist
        /private/tmp/my-app/node_modules/css-loader doesn't exist
      /private/tmp/node_modules doesn't exist or is not a directory
      /private/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
ModuleNotFoundError: Module not found: Error: Can't resolve 'css-loader' in '/private/tmp/my-app'
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/Compilation.js:1506:33
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:644:18
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:10:1)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:197:20
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:9:1)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:404:22
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:60:14
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:476:15
    at /private/tmp/my-app/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:694:7
    at done (/private/tmp/my-app/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:771:13)
    at /private/tmp/my-app/node_modules/.pnpm/webpack@5.76.1/node_modules/webpack/lib/NormalModuleFactory.js:831:22
    at finishWithoutResolve (/private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:193:14)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:258:22
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:27:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:72:26
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:15:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:42:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/ConditionalPlugin.js:40:20
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/Resolver.js:301:9
    at eval (eval at create (/private/tmp/my-app/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:28:14), <anonymous>:16:1)
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/forEachBail.js:11:16
    at /private/tmp/my-app/node_modules/.pnpm/enhanced-resolve@5.12.0/node_modules/enhanced-resolve/lib/ModulesInHierarchicalDirectoriesPlugin.js:59:22
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
    at runNextTicks (node:internal/process/task_queues:65:3)
    at processImmediate (node:internal/timers:437:9)
resolve 'css-loader' in '/private/tmp/my-app'
  Parsed request is a module
  using description file: /private/tmp/my-app/package.json (relative path: .)
    resolve as module
      looking for modules in /private/tmp/my-app/node_modules
        single file module
          using description file: /private/tmp/my-app/package.json (relative path: ./node_modules/css-loader)
            no extension
              /private/tmp/my-app/node_modules/css-loader doesn't exist
            .js
              /private/tmp/my-app/node_modules/css-loader.js doesn't exist
        /private/tmp/my-app/node_modules/css-loader doesn't exist
      /private/tmp/node_modules doesn't exist or is not a directory
      /private/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

✔ Would you like to help improve Storybook by sending anonymous crash reports? … yes
 ELIFECYCLE  Command failed with exit code 1.
```

### System

```shell
Environment Info:

  System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
  Browsers:
    Chrome: 111.0.5563.64
    Firefox: 110.0.1
    Safari: 16.3
  npmPackages:
    @storybook/addon-essentials: ^7.0.0-rc.3 => 7.0.0-rc.3 
    @storybook/addon-interactions: ^7.0.0-rc.3 => 7.0.0-rc.3 
    @storybook/addon-links: ^7.0.0-rc.3 => 7.0.0-rc.3 
    @storybook/blocks: ^7.0.0-rc.3 => 7.0.0-rc.3 
    @storybook/nextjs: ^7.0.0-rc.3 => 7.0.0-rc.3 
    @storybook/react: ^7.0.0-rc.3 => 7.0.0-rc.3 
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1
```


### Additional context

_No response_

--- a/code/frameworks/nextjs/src/css/webpack.ts
+++ b/code/frameworks/nextjs/src/css/webpack.ts
@@ -15,9 +15,9 @@ export const configureCss = (baseConfig: WebpackConfig, nextConfig: NextConfig):
       rules[i] = {
         test: /\.css$/,
         use: [
-          'style-loader',
+          require.resolve('style-loader'),
           {
-            loader: 'css-loader',
+            loader: require.resolve('css-loader'),
             options: {
               importLoaders: 1,
               ...getImportAndUrlCssLoaderOptions(nextConfig),
@@ -27,7 +27,7 @@ export const configureCss = (baseConfig: WebpackConfig, nextConfig: NextConfig):
               },
             },
           },
-          'postcss-loader',
+          require.resolve('postcss-loader'),
         ],
       };
     }
@@ -35,19 +35,19 @@ export const configureCss = (baseConfig: WebpackConfig, nextConfig: NextConfig):
   rules?.push({
     test: /\.(scss|sass)$/,
     use: [
-      'style-loader',
+      require.resolve('style-loader'),
       {
-        loader: 'css-loader',
+        loader: require.resolve('css-loader'),
         options: {
           importLoaders: 3,
           ...getImportAndUrlCssLoaderOptions(nextConfig),
           modules: { auto: true, getLocalIdent: getCssModuleLocalIdent },
         },
       },
-      'postcss-loader',
-      'resolve-url-loader',
+      require.resolve('postcss-loader'),
+      require.resolve('resolve-url-loader'),
       {
-        loader: 'sass-loader',
+        loader: require.resolve('sass-loader'),
         options: {
           sourceMap: true,
           sassOptions: nextConfig.sassOptions,

